onnxruntime/core/providers/openvino/ov_interface.cc

```diff
@@ -277,35 +352,190 @@ void OVInferRequest::StartAsync() {
   try {
     ovInfReq.start_async();
   } catch (const Exception& e) {
-    ORT_THROW(log_tag + " Couldn't start Inference: " + e.what());
+    throw std::runtime_error(log_tag + " Couldn't start Inference: " + e.what());
   } catch (...) {
-    ORT_THROW(log_tag + " In Error Couldn't start Inference");
+    throw std::runtime_error(log_tag + " In Error Couldn't start Inference");
   }
 }
 
 void OVInferRequest::Infer() {
   try {
     ovInfReq.infer();
   } catch (const Exception& e) {
-    ORT_THROW(log_tag + " Couldn't start Inference: " + e.what());
+    throw std::runtime_error(log_tag + " Couldn't start Inference: " + e.what());
   } catch (...) {
-    ORT_THROW(log_tag + " In Error Couldn't start Inference");
+    throw std::runtime_error(log_tag + " In Error Couldn't start Inference");
   }
 }
 
 void OVInferRequest::WaitRequest() {
   try {
     ovInfReq.wait();
   } catch (const Exception& e) {
-    ORT_THROW(log_tag + " Wait Model Failed: " + e.what());
+    throw std::runtime_error(log_tag + " Wait Model Failed: " + e.what());
   } catch (...) {
-    ORT_THROW(log_tag + " Wait Mode Failed");
+    throw std::runtime_error(log_tag + " Wait Mode Failed");
   }
 }
 
 void OVInferRequest::QueryStatus() {
   std::cout << "ovInfReq.query_state()"
             << " ";
 }
+
+StatefulOVInferRequest::StatefulOVInferRequest(ov::InferRequest infer_request, std::string device)
+    : OVInferRequest(std::move(infer_request)), target_device(device) {
+  bool gpu_or_npu = ((device.find("NPU") != std::string::npos) || (device.find("GPU") != std::string::npos));
+  if (gpu_or_npu) {
+    prefill_use_full_chat_history = true;
+  }
+}
+
+void StatefulOVInferRequest::CacheTensor(const std::string& tensor_name, const ov::element::Type& type,
+                                         const std::vector<size_t>& shape, int32_t fill_value) {
+  ov::Tensor tensor = ov::Tensor(type, shape);
+  std::fill_n(tensor.data<int32_t>(), tensor.get_size(), fill_value);
+  ovInfReq.set_tensor(tensor_name, tensor);
+}
```

Comment:
Should this be called 'FillTensor'? It doesn't seem to *cache* anything, right?

Reply:
fixed